### PR TITLE
Geometry Clang Check modernize-use-emplace

### PR DESCRIPTION
**Pull request by ianna**
Geometry part of @davidlange6's PR https://github.com/cms-sw/cmssw/pull/19654

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
@kpedro88 - please, review

**Comment by ianna** [comment]
@kpedro88 - what about this one?

File: Geometry/HcalTowerAlgo/src/HcalFlexiHardcodeGeometryLoader.cc [hunk 1]
```diff
@@ -86,7 +86,7 @@ std::vector<HcalFlexiHardcodeGeometryLoader::HBHOCellParameters> HcalFlexiHardco
 		  << "]: " << etabins[i].layer[k].first-1 << ":"
 		  << etabins[i].layer[k].second << std::endl;
 #endif
-	result.push_back (HcalFlexiHardcodeGeometryLoader::HBHOCellParameters(iring, depth, etabin.phis[j].first, etabin.phis[j].second, dphi, rmin, rmax, etabin.etaMin, etabin.etaMax));
+	result.emplace_back (HcalFlexiHardcodeGeometryLoader::HBHOCellParameters(iring, depth, etabin.phis[j].first, etabin.phis[j].second, dphi, rmin, rmax, etabin.etaMin, etabin.etaMax));
```
Comment (kpedro88) [inline Geometry/HcalTowerAlgo/src/HcalFlexiHardcodeGeometryLoader.cc]: should the constructor be removed here (and for the others in this file and the other loader)?

Comment (ianna) [inline Geometry/HcalTowerAlgo/src/HcalFlexiHardcodeGeometryLoader.cc]: IMHO, it's better for readability to keep it - there are way too many parameters in the constructor

**Comment by kpedro88** [comment]
+1

**Pull request merged**